synapse/federation/send_queue.py

```diff
@@ -250,32 +255,30 @@ def federation_ack(self, token):
         self._clear_queue_before_pos(token)
 
     async def get_replication_rows(
-        self, from_token, to_token, limit, federation_ack=None
-    ):
+        self, from_token: int, to_token: int, target_row_count: int
+    ) -> Tuple[List[Tuple[int, Tuple]], int, bool]:
```

Comment:
I think this return type is `StreamUpdateResult`.

Reply:
so it is.

Reply:
ah I remember now. I wasn't convinced I wanted to add an import from the depths of `synapse.replication` here. I guess I could protect it with `if TYPE_CHECKING`, but ... I did this.

Reply:
Ah, I suppose that makes sense, maybe we need a `synapse.replication.types` or something if these types are appearing outside of the bowels of the replication code. 😄 

Anyway -- not a requirement, if you think it is fine as is that's OK!